File: reversals.rb
# permanetly change the array below:

words = %w(first second third fourth fifth sixth)

# so that the array is reversed and the individual words are reversed
# ['htxis', 'htfif', 'htruof', 'driht', 'dnoces', 'tsrif']


# this solves the map! problem since it doesn't produce a new array
words.reverse!.each do |word|
  word.reverse!
end


# this is not desired since: map produces a new array so it changes the original array as well as produce a new array
words.reverse!.map do |word|
  word.reverse!
end

# used map instead of each to recreate an array

# another way would be
words.reverse.map! do |word|
  word.reverse!
end

File: reversals.js
var words = ['first', 'second', 'third', 'fourth', 'fifth'];

//reverse the array and letters

words.reverse();

for (var i=0; i < words.length; i++) {

  words[i] = words[i].split("").reverse().join("");
}

console.log(words);

var words = ['first', 'second', 'third', 'fourth', 'fifth'];

// without reverse
// create a new array
var result = [];
// create a temp variable for later
var tempWord = "";
// loop over word list from the end
for (var i = words.length -1; i >= 0; i--) {
  // loop over letters in the word from the end
  for (var j = words[i].length - 1; j >= 0; j--) {
    // add current letter to our temp variable
    tempWord += words[i][j];
  }
  // put reversed word into array
  result.push(tempWord);
  // clears temp word
  tempWord = "";
}
// replace words array with the new result
words = result;

console.log(words);
